gui-backend/src/main/java/com/rln/gui/backend/ods/InMemoryTransferProposalRepository.java

```diff
@@ -18,6 +19,21 @@
 public class InMemoryTransferProposalRepository implements TransferProposalRepository {
   private final Collection<TransferProposal> proposals = new ConcurrentLinkedQueue<>();
 
+  @Override
+  public void update(Predicate<TransferProposal> filter,
+      Function<TransferProposal, TransferProposal> remap) {
+    var affectedProposals = proposals
+        .stream()
+        .filter(filter)
+        .collect(Collectors.toList());
+    var newProposals = affectedProposals
+        .stream()
+        .map(remap)
+        .collect(Collectors.toList());
+    proposals.removeAll(affectedProposals);
```

Comment:
FYI: if I'm not wrong, this is O^2.
One could work with HashSets here.

Reply:
Are you suggesting changing `ConcurrentLinkedQueue` to something else?

Reply:
(we first thought about adding a setter but @tamaskalcza-da came up with probable concurrency problems so we went on with this remove all / (re)add all solution)

Reply:
If you create a Set (with `Collectors.toSet()`) for `affectedProposals`, it becomes linear (somewhere in the neighborhood of `O(n+m)`)

Reply:
Are you sure?
I would think it should still iterate over affectedProposals and then for each element do the removal which is probably O(n) (as `proposals` is `ConcurrentLinkedQueue`).

Reply:
I also thought this is a series of linear searches, rather than something exponential. Am I missing something?

Reply:
Not 100% sure but the way I understood it: 
Internally `removeAll` does a `bulkRemove` which is essentially an interator over the LinkedList and for each element, there is a lookup in the provided collection. If that collection is a Set, the lookup will be O(1); hence, the entire removeAll is `O(n) with n = LinkedList size`.

Reply:
ah, you mean because of the removeAll. I see, fair point

Reply:
Yeah, but my question is how does that change with HashSet (the O(n*m))? :)

Reply:
ah you iterate over the linkedListQueueOtherWhatever and then do O(1ish) lookups?
Okay, understood.